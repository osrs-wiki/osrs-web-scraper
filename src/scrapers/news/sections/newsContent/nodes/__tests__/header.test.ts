import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import headerParser from "../header";

describe("header node", () => {
  test("h1 tag should parse and render as level 2 header", () => {
    const root = parse("<h1>Test Header</h1>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("h2 tag should parse and render as level 2 header", () => {
    const root = parse("<h2>Test Header</h2>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("h3 tag should parse and render as level 3 header", () => {
    const root = parse("<h3>Test Header</h3>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("h4 tag should parse and render as level 4 header", () => {
    const root = parse("<h4>Test Header</h4>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("h5 tag should parse and render as bold text", () => {
    const root = parse("<h5>Test Header</h5>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("h6 tag should parse and render as italic text", () => {
    const root = parse("<h6>Test Header</h6>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("h1 with bold content should parse correctly", () => {
    const root = parse("<h1>Test with <b>bold</b> text</h1>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("h2 with italic content should parse correctly", () => {
    const root = parse("<h2>Test with <i>italic</i> text</h2>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("h3 with complex content should parse correctly", () => {
    const root = parse("<h3>Test with <b>bold</b> and <i>italic</i></h3>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("h4 with link should parse correctly", () => {
    const root = parse(
      "<h4>Test with <a href='https://example.com'>link</a></h4>"
    );
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("h5 with complex content should parse correctly", () => {
    const root = parse("<h5>Test with <i>italic</i> and <b>bold</b></h5>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("h6 with complex content should parse correctly", () => {
    const root = parse("<h6>Test with <b>bold</b> nested content</h6>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("Empty h1 should handle correctly", () => {
    const root = parse("<h1></h1>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("Empty h5 should handle correctly", () => {
    const root = parse("<h5></h5>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("Empty h6 should handle correctly", () => {
    const root = parse("<h6></h6>");
    const builder = new MediaWikiBuilder();
    builder.addContents([headerParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
