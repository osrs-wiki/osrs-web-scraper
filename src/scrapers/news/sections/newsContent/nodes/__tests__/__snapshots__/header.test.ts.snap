// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header node Empty h1 should handle correctly 1`] = `
"====
"
`;

exports[`header node Empty h5 should handle correctly 1`] = `
"
"
`;

exports[`header node Empty h6 should handle correctly 1`] = `
"
"
`;

exports[`header node h1 tag should parse and render as level 2 header 1`] = `
"==Test Header==
"
`;

exports[`header node h1 with bold content should parse correctly 1`] = `
"==Test with '''bold''' text==
"
`;

exports[`header node h2 tag should parse and render as level 2 header 1`] = `
"==Test Header==
"
`;

exports[`header node h2 with italic content should parse correctly 1`] = `
"==Test with ''italic'' text==
"
`;

exports[`header node h3 tag should parse and render as level 3 header 1`] = `
"===Test Header===
"
`;

exports[`header node h3 with complex content should parse correctly 1`] = `
"===Test with '''bold''' and ''italic''===
"
`;

exports[`header node h4 tag should parse and render as level 4 header 1`] = `
"====Test Header====
"
`;

exports[`header node h4 with link should parse correctly 1`] = `
"====Test with [https://example.com link]====
"
`;

exports[`header node h5 tag should parse and render as bold text 1`] = `
"'''Test Header'''
"
`;

exports[`header node h5 with complex content should parse correctly 1`] = `
"'''Test with ''italic'' and '''bold''''''
"
`;

exports[`header node h6 tag should parse and render as italic text 1`] = `
"''Test Header''
"
`;

exports[`header node h6 with complex content should parse correctly 1`] = `
"''Test with '''bold''' nested content''
"
`;
